lib/http-api.js

```diff
@@ -296,6 +296,15 @@ module.exports.MatrixHttpApi.prototype = {
         return this._request(callback, method, fullUri, queryParams, data);
     },
 
+    redirect: function(path, queryParams, prefix) {
+        var queryString = "";
+        if (queryParams) {
+            queryString = "?" + utils.encodeParams(queryParams);
+        }
+        var fullUri = this.opts.baseUrl + prefix + path + queryString;
+        window.location.href = fullUri;
```

Comment:
The JS-SDK can run in any JS environment, including those without a browser. Please delegate responsibility for doing the redirect to React SDK.